decompiler/pipeline/controlflowanalysis/variable_name_generation.py

```diff
@@ -0,0 +1,123 @@
+import re
+from enum import Enum
+from typing import List, Optional
+
+from decompiler.pipeline.stage import PipelineStage
+from decompiler.structures.ast.ast_nodes import CaseNode, CodeNode, ConditionNode, LoopNode, SwitchNode
+from decompiler.structures.ast.syntaxtree import AbstractSyntaxTree
+from decompiler.structures.pseudo import CustomType, DataflowObject, Float, GlobalVariable, Integer, Pointer, Type, Variable
+from decompiler.task import DecompilerTask
+
+
+def _get_var_counter(var_name: str) -> Optional[str]:
+    if counter := re.match(r".*?([0-9]+)$", var_name):
+        return counter.group(1)
+    return None
+
+
+def _get_containing_variables(dfo: DataflowObject) -> List[Variable]:
+    """Returns a list of variables contained in this dataflow object"""
+    variables: List[Variable] = []
+    for sub_exp in dfo.subexpressions():
+        if isinstance(sub_exp, Variable):
+            variables.append(sub_exp)
+    return variables
+
+
+class NamingConvention(str, Enum):
+    default = "default"
+    system_hungarian = "system_hungarian"
+
+
+class VariableNameGeneration(PipelineStage):
+    """ 
+    Pipelinestage in charge of renaming variables to a configured format.
+    Currently only the 'default' or 'hungarian' system are supported.
+    """
+
+    name : str = "variable-name-generation"
+    type_prefix = {
+        Float: {16: "h", 32: "f", 64: "d", 80: "ld", 128: "q", 256: "o"},
+        Integer: {8: "ch", 16: "s", 32: "i", 64: "l", 128: "i128"},
+    }
+
+
+    def __init__(self):
+        self._ast: Optional[AbstractSyntaxTree] = None
+        self._notation: Optional[str] = None
+        self._var_name: Optional[str] = None
+        self._pointer_base: bool = True
+        self._type_separator: str = ""
+        self._counter_separator: str = ""
+        self._variables: List[Variable] = []
+        self._params : List[Variable] = []
+
+
+    def run(self, task: DecompilerTask):
+        self._ast = task.syntax_tree
+        self._notation = task.options.getstring(f"{self.name}.notation", fallback="default")
+        self._var_name = task.options.getstring(f"{self.name}.variable_name", fallback="Var")
+        self._pointer_base = task.options.getboolean(f"{self.name}.pointer_base", fallback=True)
+        self._type_separator = task.options.getstring(f"{self.name}.type_separator", fallback="")
+        self._counter_separator = task.options.getstring(f"{self.name}.counter_separator", fallback="")
+        self._params = task.function_parameters
+
+        if self._notation != NamingConvention.default:
+            self._collect()
+            self._purge()
+            self._rename()
```

Comment:
this may be the wierdest hidden subclass I've ever seen.
It seems like you want to use a MixIn here. I'd propose creating a "RenamingScheme" base class with two derivates to keep things more generic.